lib/vnet/daemon/service_darwin.m

```diff
@@ -106,6 +108,10 @@ - (void)startVnet:(VnetConfig *)vnetConfig completion:(void (^)(NSError *error))
     _dnsAddr = @(vnetConfig->dns_addr);
     _homePath = @(vnetConfig->home_path);
 
+    NSXPCConnection *currentConn = [NSXPCConnection currentConnection];
+    _egid = [currentConn effectiveGroupIdentifier];
+    _euid = [currentConn effectiveUserIdentifier];
```

Comment:
obj C is so weird man, good job figuring this out

Reply:
Honestly, once you get past the syntax it's not that different from other OO languages. :D